python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,299 @@
+            self.white_light_source = salobj.Remote(
+                domain=self.domain,
+                name="ATWhiteLight",
+            )
+
+        if self.monochromator is None:
+            self.monochromator = salobj.Remote(
+                domain=self.domain,
+                name="ATMonochromator",
+            )
+
+        await asyncio.gather(
+            self.white_light_source.start_task,
+            self.monochromator.start_task,
+        )
+
+        self.log.info("Configure completed")
+
+    def set_metadata(self, metadata):
+        """Compute estimated duration.
+
+        Parameters
+        ----------
+        metadata : SAPY_Script.Script_logevent_metadataC
+        """
+        metadata.duration = self.timeout_chiller_cool_down + self.timeout_lamp_warm_up
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        await self.checkpoint("Starting chiller")
+        await self.start_chiller()
+
+        await self.checkpoint("Waiting for chiller to cool to set temperature")
+        await self.wait_for_chiller_temp_within_tolerance()
+
+        await self.checkpoint("Opening the shutter")
+        await self.open_white_light_shutter()
+
+        await self.checkpoint("Turning on lamp")
+        await self.switch_lamp_on()
+
+        await self.checkpoint("Waiting for lamp to warm up")
+        await self.wait_for_lamp_to_warm_up()
+
+        await self.checkpoint("Configuring ATMonochromator")
+        await self.configure_atmonochromator()
+
+    async def start_chiller(self):
+        await self.white_light_source.cmd_setChillerTemperature.set_start(
+            temperature=self.chiller_temperature, timeout=self.cmd_timeout
+        )
+        await self.white_light_source.cmd_startChiller.start(
+            timeout=self.timeout_chiller_cool_down
+        )
+
+    async def wait_for_chiller_temp_within_tolerance(self):
+        start_chill_time = time.time()
+        while time.time() - start_chill_time < self.timeout_chiller_cool_down:
+            chiller_temps = await self.white_light_source.tel_chillerTemperatures.next(
+                flush=True, timeout=self.timeout_std
+            )
+            tel_chiller_temp = chiller_temps.supplyTemperature
+            self.log.debug(
+                f"Chiller supply temperature: {tel_chiller_temp:0.1f} deg "
+                f"[set:{chiller_temps.setTemperature} deg]."
+            )
+            if (
+                abs(chiller_temps.setTemperature - tel_chiller_temp)
+                / chiller_temps.setTemperature
+                <= self.chiller_temp_tolerance_relative
+            ):
+                chill_time = time.time() - start_chill_time
+                self.log.info(
+                    f"Chiller reached target temperature, {tel_chiller_temp:0.1f} deg"
+                    f"within tolerance in {chill_time:0.1f} s."
+                )
+                break
+            else:
```

Comment:
I think this `else` statement is moot, you can probably remove it (and remove `continue` as well).  Usually we do this if there's anything else that needs to be done in the loop and we want to escape, however this is not the case here as there's nothing else below it.